src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,205 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "Domain/Creators/DomainCreator.hpp"  // IWYU pragma: keep
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+namespace creators {
+
+/*!
+ * \ingroup DomainCreatorsGroup
+ *
+ * \brief A general domain for two compact objects.
+ *
+ * \image html binary_compact_object_domain.png "A BHNS domain."
+ *
+ * Creates a 3D Domain consisting of four/five layers of blocks:
+ * - 0: The block at the center of each compact object, if not excised.
+ * - 1: The first layer, has the blocks for each compact object.
+ * - 2: The second layer, has the blocks that surround each object with a cube.
+ * - 3: The third layer, has the blocks that surround each cube with a
+ * half-cube.
+ * - 4: The fourth layer is made of the blocks that form the outer sphere.
```

Comment:
I found the above portion somewhat terse, and in particular I wasn't clear about: why the number of layers can be 4 or 5, what the configuration of the layers is, and what the shape of each layer is. I suggest some rephrasings to clarify what I found unclear (hopefully my interpretation is correct), and to streamline a few sentences:

```
  * Creates a 3D Domain that represents a binary compact object solution. The
  * Domain consists of four/five nested layers of blocks; these layers are,
  * working from the interior toward the exterior:
  * - 0: (optionally) The block at the center of each compact object, if not
  *      excised. If present, this block is a cube. If excised, the hole left
  *      by its absence is spherical.
  * - 1: The blocks that resolve each individual compact object. This layer has
  *      a spherical outer boundary - if the corresponding layer-0 block exists,
  *      then the layer is a cube-to-sphere transition; if the layer-0 block is
  *      excised, then the layer is a spherical shell.
  * - 2: The blocks that surround each object with a cube. Around each compact
  *      object, this layer transitions from a sphere to a cube.
  * - 3: The blocks that surround each cube with a half-cube. At this layer, the
  *      two compact objects are enclosed in a single cube-shaped grid.
  * - 4: The blocks that form the outer sphere. This layer transitions back to
  *      spherical and can extent to large radial distances from the compact
  *      objects.```